SOLIDHomework.Core/OrderService.cs

```diff
@@ -16,76 +12,97 @@ namespace SOLIDHomework.Core
     //think about SRP, DI, OCP
     //maybe for each type of payment type make sense to have own Order-based class?
     public class OrderService
```

Comment:
- [ ] 1. In order to follow Single Responsibility principle, we need to main a single Idea\Responsibility in each class. Let's take OrderService class as an example: The purpose of this class is to checkout or process the order. Now, keeping that in mind, do we need to have an additional classes/logic here?
- [ ] 2. Let's avoid nested classes. If you have a class\interface and etc., create a separate .cs file for it. The exception will be a Constants static class with nested specific classes.
Example:
```
public static class Constants()
{
    public static class CommonConstants()
    {}
    public static class AzureConstants()
    {}
    public static class DomainConstants()
    {}
 }
 ```
- [ ] 3. Let's follow a folder structure. Consider putting classes/interfaces in the correlated folders.
Example:
If we have a service class, put it in the Services folder.
If we have an exception class, put it in the Exceptions folder.
- [ ] 4. In order to follow Dependency Inversion principle, we need to maintain abstractions dependencies for both high and low level modules. 
[Example (click on me)](https://www.c-sharpcorner.com/article/solid-principles-in-c-sharp-dependency-inversion-principle/)